src/main/java/com/arangodb/async/internal/ArangoExecutorAsync.java

```diff
@@ -70,8 +71,12 @@ public ArangoExecutorAsync(final VstCommunicationAsync communication, final Aran
                 .thenApplyAsync(responseDeserializer::deserialize);
     }
 
-    public void disconnect() throws IOException {
-        communication.close();
-        outgoingExecutor.shutdown();
+    public void disconnect() {
+        try {
+            communication.close();
```

Comment:
Shouldn't  `communication.close()` be wrapped with a separate `try-catch` as well?

Reply:
right, I moved the executor service shutdown inside a finally block